### ISSUE TITLE
Where is the tree structure actually generated?

### ISSUE THREAD

**Issue by bakhtiarZ** [LangProc/langproc-2021-cw#17]
In lab2, we have many classes and derived classes, this seems like a linked list type structure and it is shown to create an AST but I can't find where it does this, like the actual linking of nodes and creation of the tree.

Is it done in the parser where we do things like $$ = $1 ? Is this linking the node to the previous one from the grammar>?

**Comment by ymherklotz** [comment]
Yes, the parser is the one that generates the AST. You do that by allocating new nodes wherever they are needed, and otherwise just assigning the pointer that was already constructed by one of the nodes lower down in the parser returned (`$$ = $1`).

I wouldn't call it a linked list though, it's just a tree structure, with various different nodes that link to more nodes.

**Comment by bakhtiarZ** [comment]
Ok cool so I would have to just make the classes and the grammar hierarchy
sorts out the structure of the tree?
